data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/SqsWorkerTest.java

```diff
@@ -245,4 +247,30 @@ void processSqsMessages_should_invoke_delete_if_input_is_not_valid_JSON_and_dele
         verify(sqsMessagesDeletedCounter).increment(1);
         verify(sqsMessagesFailedCounter).increment();
     }
+
+    @Test
+    void populateS3Reference_should_interact_with_getUrlDecodedKey() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
```

Comment:
Can we not test an s3Entity with a key that has a space in it? Why do we need reflection?